src/providers/FileSystemProvider/TextSearchProvider.ts

```diff
@@ -104,13 +104,51 @@ export class TextSearchProvider implements vscode.TextSearchProvider {
     } else {
       const sysStr = params.has("system") && params.get("system").length ? params.get("system") : "0";
       const genStr = params.has("generated") && params.get("generated").length ? params.get("generated") : "0";
+
+      let uri = options.folder;
+
+      if (!params.get("filter")) {
+        // Unless isfs spec already includes a non-empty filter (which it rarely does), apply includes and excludes at the server side.
+        // If include or exclude field is set to, say, A1B2M*.int there will be two consecutive options.[in|ex]cludes elements:
+        //   **/A1B2M*.int/**
+        //   **/A1B2M*.int
+        //
+        // Ignore first, and strip **/ prefix from second.
+        // When 'Use Exclude Settings and Ignore Files' is enabled (which is typical) options.excludes will also contain entries from files.exclude and search.exclude settings.
```

Comment:
Is there any way to ignore this setting or throw away those excludes? I'm testing this now and with the setting on, you get a really long value for `files` that is mostly junk:

```
files=User,User.**,'.git,'.svn,'.hg,'CVS,'.DS_Store,'Thumbs.db,'node_modules,'bower_components,'*.code-search,'*.prj
```

All of these extra file excludes do nothing and since this is all passed in the URL and not the body, we could run into issues with the URL string being too long. My search include is just the User folder in an `isfs` folder with no `type` or `filter`, so the `files` that I'd expect to see is `User.*.cls, User.*.mac,User.*.int,User.*.inc,'*.prj`. You need the 4 extensions because otherwise all file types will be searched, and not all file types are in the tree. Sorry to be nitpicky but this is more complicated than it looks at first glance.

Reply:
To strip off the excludes that originate from `files.exclude` and `search.exclude` maybe I can get these settings, dedupe them, count them, then remove that many entries from the end of the excludes array which the search API gives me.

I want to look in more detail at how to handle package stuff.